tests/test_distortions.py

```
@@ -115,15 +104,24 @@ def test_distort_V_Cd(self, mock_print):
-            self.V_Cd_struc, 0, 0.5, frac_coords=vac_coords, verbose=True
-        )
+        distortions.distort(self.V_Cd_struc, 0, 0.5, frac_coords=vac_coords, verbose=True)
         mock_print.assert_called_with(
             f"\tDefect Site Index / Frac Coords: {vac_coords}\n"
             + "            Original Neighbour Distances: []\n"
             + "            Distorted Neighbour Distances:\n\t[]"
         )
 
+    def test_distort_negative_factor(self):
+        vac_coords = np.array([0, 0, 0])  # Cd vacancy fractional coordinates
+        with self.assertRaises(ValueError) as e:
```

Ruff (0.11.9)

Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)